Changes to source files to compile Singular on SunOS.

This patch is not actually applied, only the changes to auto-generated
files in sunos.patch should be applied.


commit 6712e907767e5540613b217f792b5f42d2b39ac1
Author: Dima Pasechnik <dimpase@gmail.com>
Date:   Wed Jan 31 12:50:27 2018 +0000

    link libraries sometimes needed by gethostbyname etc
    
    On Solaris, in particular, one needs -lsocket -lnsl in the libs
    to be linked against. See https://trac.sagemath.org/ticket/24611
    This is taken care by this autoconf macro call.

diff --git a/configure.ac b/configure.ac
index ecda2a4..a4a21a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -155,6 +155,7 @@ m4_ifval([$3],
 fi[]dnl
 ])])
 
+AX_LIB_SOCKET_NSL
 
 AC_DEFINE_UNQUOTED([CC],"$CC",[CC])
 AC_DEFINE_UNQUOTED([CXX],"$CXX",[CXX])
diff --git a/m4/ax_lib_socket_nsl.m4 b/m4/ax_lib_socket_nsl.m4
new file mode 100644
index 0000000..54cad68
--- /dev/null
+++ b/m4/ax_lib_socket_nsl.m4
@@ -0,0 +1,40 @@
+# ===========================================================================
+#    https://www.gnu.org/software/autoconf-archive/ax_lib_socket_nsl.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_LIB_SOCKET_NSL
+#
+# DESCRIPTION
+#
+#   This macro figures out what libraries are required on this platform to
+#   link sockets programs.
+#
+#   The common cases are not to need any extra libraries, or to need
+#   -lsocket and -lnsl. We need to avoid linking with libnsl unless we need
+#   it, though, since on some OSes where it isn't necessary it will totally
+#   break networking. Unisys also includes gethostbyname() in libsocket but
+#   needs libnsl for socket().
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Russ Allbery <rra@stanford.edu>
+#   Copyright (c) 2008 Stepan Kasal <kasal@ucw.cz>
+#   Copyright (c) 2008 Warren Young <warren@etr-usa.com>
+#
+#   Copying and distribution of this file, with or without modification, are
+#   permitted in any medium without royalty provided the copyright notice
+#   and this notice are preserved. This file is offered as-is, without any
+#   warranty.
+
+#serial 7
+
+AU_ALIAS([LIB_SOCKET_NSL], [AX_LIB_SOCKET_NSL])
+AC_DEFUN([AX_LIB_SOCKET_NSL],
+[
+	AC_SEARCH_LIBS([gethostbyname], [nsl])
+	AC_SEARCH_LIBS([socket], [socket], [], [
+		AC_CHECK_LIB([socket], [socket], [LIBS="-lsocket -lnsl $LIBS"],
+		[], [-lnsl])])
+])
commit facbf5bceffaab92df81e58b805ceee5cdc1788e
Author: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date:   Tue Jan 30 17:38:13 2018 +0100

    fix: tr. #816 (P_PROCS_MODULE_LDFLAGS)

diff --git a/libpolys/polys/Makefile.am b/libpolys/polys/Makefile.am
index 23d4cae..b222a7f 100644
--- a/libpolys/polys/Makefile.am
+++ b/libpolys/polys/Makefile.am
@@ -65,7 +65,7 @@ p_Procs_FieldIndep_la_CPPFLAGS = -Dp_Procs_FieldIndep ${P_PROCS_CPPFLAGS_COMMON}
 p_Procs_FieldQ_la_CPPFLAGS = -Dp_Procs_FieldQ ${P_PROCS_CPPFLAGS_COMMON}
 p_Procs_FieldZp_la_CPPFLAGS = -Dp_Procs_FieldZp ${P_PROCS_CPPFLAGS_COMMON}
 
-P_PROCS_MODULE_LDFLAGS = -shared -module -dynamic -export-dynamic -avoid-version -weak_reference_mismatches weak -undefined dynamic_lookup -Wl,-undefined -Wl,dynamic_lookup -flat_namespace $(SINGULAR_LDFLAGS)
+P_PROCS_MODULE_LDFLAGS = -shared -module -dynamic -export-dynamic -avoid-version -weak_reference_mismatches weak -undefined dynamic_lookup -flat_namespace
 
 p_Procs_FieldGeneral_la_LDFLAGS = ${P_PROCS_MODULE_LDFLAGS}
 p_Procs_FieldIndep_la_LDFLAGS = ${P_PROCS_MODULE_LDFLAGS}
